eo-runtime/src/main/java/org/eolang/PhDefault.java

```diff
@@ -354,8 +329,202 @@ private static void debug(final String msg) {
+            this.attr = attr;
+            this.name = name;
+            this.phi = phi;
+        }
+
+        /**
+         * Reproduce NamedAttr as {@link AtNamed}.
+         *
+         * @return Named attr
+         */
+        public AtNamed asAtNamed() {
+            return new AtNamed(
+                String.format(
+                    "%s#%s",
+                    this.phi.getClass().getCanonicalName(),
+                    this.name
+                ),
+                String.format(
+                    "%s.%s",
+                    new Oname(this.phi).asString(),
+                    this.name
+                ),
+                this.phi,
+                this.attr
+            );
+        }
+    }
+
+    /**
+     * Get object name, as in source code.
+     *
+     * @since 0.1
+     */
+    private static final class Oname {
+
+        /**
+         * Object to get name.
+         */
+        private final Phi phi;
+
+        /**
+         * Ctor.
+         *
+         * @param phi The phi
+         */
+        Oname(final Phi phi) {
+            this.phi = phi;
+        }
+
+        /**
+         * Object source code name.
+         *
+         * @return Name as string
+         */
+        private String asString() {
+            String txt = this.phi.getClass().getSimpleName();
+            final XmirObject xmir = this.getClass().getAnnotation(XmirObject.class);
+            if (null != xmir) {
+                txt = xmir.oname();
+                if ("@".equals(txt)) {
+                    txt = "φ";
+                }
+            }
+            return txt;
+        }
+    }
+
+    /**
+     * The map of attrs which was extended.
+     *
+     * @since 0.1
+     */
+    private static final class ExtendedAttrs {
+
+        /**
+         * The name of new attr.
+         */
+        private final String name;
+
+        /**
+         * The value of new attr.
+         */
+        private final Attr attr;
+
+        /**
+         * Attrs which be extended.
+         */
+        private final Attrs attrs;
+
+        /**
+         * Ctor.
+         *
+         * @param name The name of new attr
+         * @param attr The value of new attr
+         * @param attrs Attrs which be extended
+         */
+        ExtendedAttrs(final String name, final Attr attr, final Attrs attrs) {
+            this.name = name;
+            this.attr = attr;
+            this.attrs = attrs;
+        }
+
+        /**
+         * All attrs.
+         *
+         * @return Attrs as map
+         */
+        public Map<String, Attr> asMap() {
+            if (PhDefault.SORTABLE.matcher(this.name).matches()) {
+                this.attrs.orderAsList().add(this.name);
+            }
+            this.attrs.asMap().put(this.name, this.attr);
+            return this.attrs.asMap();
+        }
+    }
+
+    /**
+     * Multiple attrs.
+     *
+     * @since 0.1
+     */
+    private static final class Attrs {
```

Comment:
@l3r8yJ This class seems to be plain data holder. It performs no logic.
See: https://www.yegor256.com/2016/07/06/data-transfer-object.html

Reply:
@mximp Yes, I agree that this is a bad design, but without it we would have 4 args in the `ExtendedAttrs` ctor, and qulice does not pass them without the `@checkstyle` tag, what will be better here `@checkstyle` tag or this dto?